code/datums/mind.dm

```diff
@@ -449,157 +438,92 @@
 					if(1)
 						target_antag = antag_datums[1]
 					else
-						var/datum/antagonist/target = input("Which antagonist gets the objective:", "Antagonist", def_value) as null|anything in antag_datums + "(new custom antag)"
+						var/datum/antagonist/target = input("Which antagonist gets the objective:", "Antagonist", "(new custom antag)") as null|anything in antag_datums + "(new custom antag)"
 						if (QDELETED(target))
 							return
 						else if(target == "(new custom antag)")
 							target_antag = add_antag_datum(/datum/antagonist/custom)
 						else
 							target_antag = target
 
-		var/new_obj_type = input("Select objective type:", "Objective type", def_value) as null|anything in list("assassinate", "maroon", "debrain", "protect", "destroy", "prevent", "hijack", "escape", "survive", "martyr", "steal", "download", "nuclear", "capture", "absorb", "custom")
-		if (!new_obj_type)
+		var/list/allowed_types = list(

```

Comment:
typecacheof(/datum/objective, TRUE), then use typecache everywhere else?

Reply:
This is not all objective types, only the ones that support admin editing.